### DISCUSSION TITLE
Checking if a some value exists across many arrays which are selected as input.

### DISCUSSION THREAD

**Discussion by avosepp** [open-policy-agent/community#446] [OPA and Rego] [answered]
Hello! 

I have a data structure which is a map of arrays. 

`{
  "key1": ["string1", "string2"],
  "key2": ["string3", "string4"],
  "key3": ["string1", "string3"]
}`

I get an input which is an array of possible keys from the map. 
`data.input.keyInput = ["key1", "key3"]`

I have a final data structure which contains an array of strings, which may or may not, exist under the keys. 

`data.stringValues = ["string3", "string4"]`

The rule I'm looking to write should return true if `some` string in `stringValues` exists in any array whose key was input as `keyInput`. 

So for this example, the rule would return true, because `"string3"` exists under `"key3"`, and `"key3"` was an input provided by the user. 

I'm new to Rego and not sure how I should structure this. I considered performing `array.concat` to combine all the keys under one array. But because I can get an arbitrary number of keys input, I don't know how to loop and concat all the key arrays together. 

Any ideas that could point me in the right direction? Much appreciated.

**Comment by anderseknert** [comment, marked as answer]
Hi @avo-sepp 👋 There are many ways to do that, but here's an example: https://play.openpolicyagent.org/p/CHr9VZ9fa7

```rego
package play

import future.keywords.contains
import future.keywords.if
import future.keywords.in

map := {"key1": ["string1", "string2"], "key2": ["string3", "string4"], "key3": ["string1", "string3"]}

matches contains match if {
    some key in input.keyInput
    some match in map[key]
    match in data.stringValues
}

```

**Reply by avosepp** [reply]
Thank you! I'm still learning and this taught me a whole lot. I got my code working!

**Reply by anderseknert** [reply]
Great! What are you planning to use OPA for? If you don't mind me asking of course 😃 Just curious!
